### DISCUSSION TITLE
Updating the fmt progress in the same line

### DISCUSSION THREAD

**Discussion by tgodzik** [scalameta/scalafmt#3636] [Ideas]
Based on https://github.com/VirtusLab/scala-cli/issues/2372

How much effort would be to make the CLI update progress inline?

![image](https://github.com/scalameta/scalafmt/assets/3807253/e44a323f-d43e-4972-a7ef-739d343cb8c5)

I wonder if maybe @kitbellew has any idea

**Comment by kitbellew** [comment]
according to [this](https://stackoverflow.com/questions/1403772/how-can-i-check-if-a-java-programs-input-output-streams-are-connected-to-a-term), we don't know if we are writing to a console, and hence it seems we can't really do that.

**Reply by tgodzik** [reply]
Hmm... there is something for that in coursier https://github.com/coursier/coursier/blob/main/modules/cache/jvm/src/main/scala/coursier/cache/loggers/ProgressBarRefreshDisplay.scala#L10

I guess that's handled by separate artifact used on command line?

**Comment by kitbellew** [comment]
i think that you're right, and a similar approach may be used in scalafmt, as there are references to alex archambault. is it possible that the original problem is either limited to that other tool, or is somehow called from a script not connected to a terminal?

**Reply by tgodzik** [reply]
> is it possible that the original problem is either limited to that other tool, or is somehow called from a script not connected to a terminal?

Do you mean how the user is running format? That is done via Scala CLI, but it uses native images of scalafmt underneath.

**Comment by kitbellew** [comment]
i meant that i don't think i can see this problem if i simply run `scalafmt` binary from a command line.

**Reply by tgodzik** [reply]
Ach right, so this is an issue on the Scala CLI side entirely, totally missed that, thanks!
